README.md

```diff
@@ -53,39 +56,76 @@ kubectl apply -f scone/sgx-dev-plugin.yaml
 kubectl apply -f scone/scone-namespace.yaml 
 kubectl apply -f scone/las-daemonset.yaml
 kubectl apply -f scone/cli-scone.yaml
+
 kubectl get pods -n scone
 ```
 
-In this demonstration, we will use a public CAS at `5-4-0.scone-cas.cf`. In order to use this CAS, we need to create a namepace to post the sessions (configuration and attestation info for confidential workloads) in a way to not overlap with other users.
+In this demonstration, we will use a public CAS at `5-6-0.scone-cas.cf`. In order to use this CAS, we need to create a namepace to post the sessions (configuration and attestation info for confidential workloads) in a way to not overlap with other users.
 
 To do this, we can create a random string to be our namespace name.
 
 ```bash
+# Create variable MY_NAMESPACE
 export MY_NAMESPACE=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 15 ; echo '')
 
+# Provide CLI address for next commands
+export SCONE_CLI_ADDRESS=$(kubectl -n scone get pods \
+  -o=jsonpath='{.items[0].metadata.name}' -l app=scone-cli)
+
 # Make $MY_NAMESPACE available inside the scone-cli container
-kubectl exec -it -n scone $(kubectl get pods -n scone -o=jsonpath='{.items[0].metadata.name}' \
-    -l app=scone-cli)  -- /bin/bash -c 'echo "export MY_NAMESPACE='$MY_NAMESPACE'" > /root/.bashrc'
+kubectl exec -it -n scone \
+  $SCONE_CLI_ADDRESS -- /bin/bash \
+  -c 'echo "export MY_NAMESPACE='$MY_NAMESPACE'" > /root/.bashrc'
+
+# Copy over scone sesison
+kubectl cp scone/scone-service-namespace.yaml \
+    scone/$SCONE_CLI_ADDRESS:scone-service-namespace.yaml
+
+# Copy over spire-ca sesison
+kubectl cp scone/scone-spire-ca-namespace.yaml \
```

Comment:
The way the templates were set, all the way back in the plugin's documentation, for instance, `spire-svid-<\session-name-selector>`  and `spire-ca-<\trust-bundle-session-name>` push towards multiple namespaces. 

Trying to simply use the selector as the session name, like `<\session-name-selector>` or `<\trust-bundle-session-name>` doesn't seem to work. I've added a commit with these updates that lead to a bad request to CAS (400). **Cas complains the session is missing a predecessor.**


Below, we have the logs for the Agent failing to post the session using my last commit:

```bash
time="2021-12-06T20:25:04Z" level=info msg="Posting Session:" external=false name: 2kRkjyCBdYMF7Ra/svid-session                                                                        
version: "0.3"                                                                                                                                                                         
predecessor: ~                                                                                                            
secrets:                                                                                                                  
  - name: svid                                                                                                            
    kind: x509                                                                                                                                                                         
    value: |                                                                                                                                                                           
        -----BEGIN CERTIFICATE-----                                                                                                                                                                        
        MIIB6jCCAZCgAwIBAgIRAMYFWPfysokzkNvDWCJzDFcwCgYIKoZIzj0EAwIwHjEL                                                                     
        MAkGA1UEBhMCVVMxDzANBgNVBAoTBlNQSUZGRTAeFw0yMTEyMDYyMDIzNTZaFw0y                                                  
        MTEyMDcwNjI0MDZaMB0xCzAJBgNVBAYTAlVTMQ4wDAYDVQQKEwVTUElSRTBZMBMG                                                  
        ByqGSM49AgEGCCqGSM49AwEHA0IABE2yceCSCQ8HnI5huJvMqfCIQbq+WO0kFsQJ                                                  
        dx4xwUAY+k6JGmw+F3cUwh9+Yk4kLapySaRK/n5h+D21RfrEszWjga8wgawwDgYD                                                  
        VR0PAQH/BAQDAgOoMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAMBgNV                                                  
        HRMBAf8EAjAAMB0GA1UdDgQWBBQ4Usttj+2iU6Tz2+ccNjKDRbKhVzAfBgNVHSME                                                  
        GDAWgBQQ1bW06MCCpVZyuRVkKp1RzHwVxDAtBgNVHREEJjAkhiJzcGlmZmU6Ly9l                                                                     
        eGFtcGxlLm9yZy9zY29uZS1zZXJ2aWNlMAoGCCqGSM49BAMCA0gAMEUCIBnEfx1K                                                  
        4caerTekaj7kR2t2pv305Dpi7PWFGG9lYWwAAiEAj/yrNgD5sqH9sf+oYFU4z8/5                                                                     
        U34NhNEzjlubVW0/29U=                                                                                              
        -----END CERTIFICATE-----                                                                                                            
                                                                                                                                             
    issuer: svid-intermediates                                                                                            
    export:                                                                                                                                  
        session: 2kRkjyCBdYMF7Ra/svid-session                                                                                                
        session_hash: f06ca49661e21a956ada5870b40da72f045e25d085b9a4f420cea853679efd35                                                       
    private_key: svid_key                                                                                                                    
  - name: svid-intermediates                                                                                                                                                           
    kind: x509-ca                                                                                                                                                                                          
    value: |                                                                                                                                                                           
        -----BEGIN CERTIFICATE-----                                                                                                                                                    
        MIIB4TCCAWagAwIBAgIQYHyvGC6ZxN29xqM7VBrWWzAKBggqhkjOPQQDAzAeMQsw                                                                                                               
        CQYDVQQGEwJVUzEPMA0GA1UECgwGU1BJRkZFMB4XDTIxMTIwNjIwMTgwMloXDTIx                                                                                                               
        MTIwNzIwMTgxMlowHjELMAkGA1UEBhMCVVMxDzANBgNVBAoTBlNQSUZGRTBZMBMG                                                                                                               
        ByqGSM49AgEGCCqGSM49AwEHA0IABBk80hTym8gb83PGxd3uNCpalHgHGBsKKius                                                                                                               
        NZBT7a5YXjF0CpLiOn7nMW+AQZLS7xNAp/DO5cZSQpwVAwKZ+9qjgYUwgYIwDgYD                                                                                                                                   
        VR0PAQH/BAQDAgGGMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFBDVtbTowIKl                                                                                                                                   
        VnK5FWQqnVHMfBXEMB8GA1UdIwQYMBaAFIel81ei8DWswPhkxFTnbtO6OcjoMB8G                                                                                                                                   
        A1UdEQQYMBaGFHNwaWZmZTovL2V4YW1wbGUub3JnMAoGCCqGSM49BAMDA2kAMGYC                                                                                
        MQCXT60ZtMJ3gChtpnqis6+QMP/D8ntq0rWz4zIwwZTRDhyWK052iCcej21I28Ku                                                                                                                                   
        4GUCMQDCL4vr7wsqfvFhZ4bOd1b6t7X0RIIRjYKQoNvoMkaoI/+W2vn2lTRnFL+Q                                                           
        KLhreHc=                                                                                                                                                                                           
        -----END CERTIFICATE-----                                                                                                                                                                          
                                                                                                                                                                                                           
    export:                                                                                                                                                                                                
        session: 2kRkjyCBdYMF7Ra/svid-session                    
        session_hash: f06ca49661e21a956ada5870b40da72f045e25d085b9a4f420cea853679efd35                                                                  
  - name: svid_key                                                                                                                 
    kind: private-key                                                                                                              
    value: |                                                                                                                                                                                                                                                          
        -----BEGIN PRIVATE KEY-----                                                                                                                                                                                                                                   
        MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQgoZjVb9CKFY8URMEZ                                                                                
        Kh+z9xSocpaEDbp/g2TRbTCTZV6hRANCAARNsnHgkgkPB5yOYbibzKnwiEG6vljt                                                           
        JBbECXceMcFAGPpOiRpsPhd3FMIffmJOJC2qckmkSv5+Yfg9tUX6xLM1                                                                   
        -----END PRIVATE KEY-----                                                                                                  
                                                                                                                                                                                                                                                                      
    export:                                                                 
        session: 2kRkjyCBdYMF7Ra/svid-session                               
        session_hash: f06ca49661e21a956ada5870b40da72f045e25d085b9a4f420cea853679efd35                                                                  
="<unknown>" plugin_name=sconecas_sessionmanager plugin_type=SVIDStore subsystem_name=catalog                                                           
time="2021-12-06T20:25:04Z" level=error msg="cannot post SVID into CAS. Response status code =" 400 Bad Request="<unknown>" external=false plugin_name=sconecas_sessionmanager plugin_type=SVIDStore subsystem_name=catalog                                                                                      
time="2021-12-06T20:25:04Z" level=error msg="cannot post SVID and its key into CAS " external=false plugin_name=sconecas_sessionmanager plugin_type=SVIDStore subsystem_name=catalog unknown predecessor hash needed for the session=2kRkjyCBdYMF7Ra/svid-session Reconfigure your CAS="<unknown>"
```

Reply:
Hi @Benardi 

> The way the templates were set, all the way back in the plugin's documentation, for instance, spire-svid-<\session-name-selector> and spire-ca-<\trust-bundle-session-name> push towards multiple namespaces.

The templates are just templates. The documentation presents a sample that you can use, but you're not obligated to. In other words, you can adjust your templates to match the individual needs of each Agent deployment.

> time="2021-12-06T20:25:04Z" level=error msg="cannot post SVID and its key into CAS " external=false plugin_name=sconecas_sessionmanager plugin_type=SVIDStore subsystem_name=catalog unknown predecessor hash needed for the session=2kRkjyCBdYMF7Ra/svid-session Reconfigure your CAS="<unknown>"

This error happens because, for some reason (e.g. volume lost), the Agent has not the predecessor for that session. From this error, the current version can not recover.